### PR TITLE
Add Subsampling Functionality

### PR DESCRIPTION
**Pull request by GregoryPetrochenkov-NOAA**
This pull request addresses #144.

## Additions

- subsampling.py:  Add subsampling functionality

## Changes

- pyproject.toml: Updated version
- requirements/base.txt: Add flox dependency
- gval_dataframe.py: Add new subsampoing dataframe creation methods
- gval_xarray.py: Add subsampling functionality
- compute_categorical_metrics.py:  Add subsampling funcitonality
- compute_continuous_metrics.py:  Add subsampling funcitonality
- categorical_statistics.py: report nan and inf stats
- continuous_statistics.py: report nan and inf stats
- schemas.py:  Add new schema for subsampling df
- cases_compute_continuous_metrics.py: Adjust for reporting nans and infs
- cases_statistics.py: Adjust for reporting nans and infs
- cases_subsampling/test_subsampling:  Tests for subsampling

### CODE REVIEW

File: src/gval/utils/exceptions.py [hunk 1]
```diff
@@ -23,3 +23,11 @@ def __init__(self):
 
     def __str__(self):
         return "Rasters don't spatially intersect."
+
+
```
Comment (fernando-aristizabal) [inline src/gval/utils/exceptions.py:27]: Still needed?

File: notebooks/Subsampling Tutorial.ipynb
Comment (fernando-aristizabal) [inline notebooks/Subsampling Tutorial.ipynb:1]: Is the `print(time...)` at the end still needed?

File: src/gval/utils/exceptions.py (hunk 1 above)
Comment (GregoryPetrochenkov-NOAA) [reply src/gval/utils/exceptions.py]: This is not and I removed it

File: notebooks/Subsampling Tutorial.ipynb
Comment (GregoryPetrochenkov-NOAA) [reply notebooks/Subsampling Tutorial.ipynb:1]: I removed the printing of time.  I also intend to make this notebooks more visually robust, but I wanted something for this PR for you to get your hands on.

**Review by fernando-aristizabal — APPROVED** [review]

**Pull request merged**